src/com/palmergames/bukkit/towny/utils/AreaSelectionUtil.java

```diff
@@ -85,34 +83,53 @@
 				} else if (owner instanceof Resident) {
 					available = TownySettings.getMaxResidentPlots((Resident) owner);
 				}
-				
-				
 
 				if (args[0].equalsIgnoreCase("auto")) {
-					// Attempt to select outwards until no town blocks remain
-
-					while (available - Math.pow((r + 1) * 2 - 1, 2) >= 0)
-						r += 1;
+					// Attempt to select outwards until no town blocks remain. 
+					// This will make a less perfect claim than /t claim rect # or /t claim circle auto.
+					r = 1;
+					int total = 0;
+					while (available - ((r * 8) + total) >= 0) { // Radius 1 grabs 8 blocks, 2 grabs 16 more requiring 24, 3 grabs 24 more requiring 48, etc...
+						total += r * 8;
+						r++;
+					}
+					
+					if (TownySettings.getMaxClaimRadiusValue() > 0) 
+						r = Math.min(r, TownySettings.getMaxClaimRadiusValue());
 
 				} else {
 					try {
 						r = Integer.parseInt(args[0]);
 					} catch (NumberFormatException e) {
 						throw new TownyException(Translation.of("msg_err_invalid_radius"));
 					}
-				}
-				if (r > TownySettings.getMaxClaimRadiusValue() && TownySettings.getMaxClaimRadiusValue() > 0) {
-					throw new TownyException(Translation.of("msg_err_invalid_radius_number", TownySettings.getMaxClaimRadiusValue()));
+					if (TownySettings.getMaxClaimRadiusValue() > 0 && r > TownySettings.getMaxClaimRadiusValue())
+						throw new TownyException(Translation.of("msg_err_invalid_radius_number", TownySettings.getMaxClaimRadiusValue()));
+
+					// Calculate how many TownBlocks are needed to claim the radius. 
+					// Start blocks at 0 if the plot is unclaimed.
+					int neededBlocks = pos.getTownBlock().hasTown() ? 0 : 1;
+					for (int i = 1; i <= r; i++)
+						neededBlocks += i * 8;
+					
+					// Rethink how much of a radius will be used, as there's not enough available TownBlocks.
+					if (neededBlocks > available) {
+						r = 1; 
+						int total = 0;
+						while (available - ((r * 8) + total) >= 0) { // Radius 1 grabs 8 blocks, 2 grabs 16 more requiring 24, 3 grabs 24 more requiring 48, etc...
+							total += r * 8;
+							r++;
+						}
+						r--; // Finally reduce the radius by 1 (so that we have a perfect ring of claims,) and replace the original r.
+					}
 				}
 					
 				if (r > 1000)
 					r = 1000;
 				for (int z = -r; z <= r; z++)
 					for (int x = -r; x <= r; x++)
-						if (out.size() < available) {
-							WorldCoord wc = new WorldCoord(pos.getWorldName(), pos.getX() + x, pos.getZ() + z);
-							if (!wc.hasTownBlock())
-								out.add(wc);
```

Comment:
This is the actual code that caused the issue with #4427, this selection tool is always assuming that you're going to be claiming land. Also below in the circle-shape selector.